questionpy_server/worker/pool.py

```diff
@@ -34,6 +34,10 @@ def __init__(self, max_workers: int, max_memory: int,
         self._semaphore: Optional[Semaphore] = None
         self._condition: Optional[Condition] = None
 
+        self._lock: Lock = Lock()
```

Comment:
Der Lock ist für diesen Fall nicht notwendig. Die Funktionen werden ja nicht in verschiedenen Threads aufgerufen und die Manipulation der Counter führt auch zu keinen Coroutine-Wechseln.